### DISCUSSION TITLE
Cognitive Load Handbook Discussion

### DISCUSSION THREAD

**Discussion by zakirullin** [zakirullin/cognitive-load#17] [Announcements]
Hi there! 

The handbook was already discussed in several places:
[It's Not That Complicated - Runtime Rundown Podcast](https://podcasters.spotify.com/pod/show/runtime-rundown/episodes/Its-Not-That-Complicated-e24rur3)
[/r/golang](https://www.reddit.com/r/golang/comments/13qazs7/cognitive_load_developers_handbook/) (Highlight: DDD)
[/r/programming](https://www.reddit.com/r/programming/comments/13rlc2p/cognitive_load_developers_handbook/) (Highlight: HTTP Statuses & Featureful Languages)
[issues](https://github.com/zakirullin/cognitive-load/issues/7) (Highlight: Onion Architecture)
  
Contributions made by the community: https://github.com/zakirullin/cognitive-load/pull/12 (credits to @erni27)

**Comment by MichaelBreeden** [comment]
Nice article...
And all us old timers are thinking "that's what I thought all along, luckily I kept doing it because it worked".
The claim that a class method should only be a few lines of code just doesn't make sense in the real world. It should be whatever length it takes to do the job. Never jump half way across a chasm because you like making small leaps. That is especially true for maintaining and debugging. 
... I could go on. DevOps... Really, a developer is supposed to keep track of all that and be able to instantly switch between numerous roles that used to belong to a number of people? No cognition problem involved there! Better still, add cloud technology and security requirements. It will make the job more interesting and fulfilling.

**Comment by zakirullin** [comment]
@MichaelBreeden Thanks for your warm feedback!

> The claim that a class method should only be a few lines of code just doesn't make sense in the real world. It should be whatever length it takes to do the job.

Exactly! There are so many rules and architectures that we've been blindly following just for the sake of it. 

Whereas in reality the answer to the question of whether or not to apply an approach is: "It depends".
